src/main/java/io/github/debuggyteam/tablesaw/integration/emi/TableSawEmiRecipe.java

```diff
@@ -17,14 +17,14 @@
 import java.util.List;
 
 public class TableSawEmiRecipe implements EmiRecipe {
-    private Identifier id;
-    private EmiIngredient input;
-    private EmiStack output;
+    private final Identifier id;
```

Comment:
Not quite sure if these are supposed to be final since idk if that would break the recipe stuff.

Reply:
they are not modified further, so that is okay

Reply:
oh ok